api/helpers/gamestate/moves/resolve/six.js

```diff
@@ -0,0 +1,50 @@
+module.exports = {
+  friendlyName: 'Resolve Six One-Off',
+
+  description: 'Returns new GameState resulting from resolving an six, which Scrap all Royals and Glasses Eights on the field.',
+
+  inputs: {
+    currentState: {
+      type: 'ref',
+      description: 'The latest GameState before the six scraps all royals and glasses eights',
+      required: true,
+    },
+  },
+  sync: true, // synchronous helper
+  fn: ({ currentState }, exits) => {
+    let result = _.cloneDeep(currentState);
+
+    const player = result.playedBy ? result.p1 : result.p0;
+    const opponent = result.playedBy ? result.p0 : result.p1;
```

Comment:
It doesn't really matter who played the six since the effect is symmetrical. And actually the `playedBy` is going to be the person who resolved, which is the opponent of the person who played the six. I think we could just grab p0 and p1 and use those instead of player and oponent like
```suggestion
   const { p0, p1 } = result;
```